scripts/mostReadCollectionTimes.js

```diff
@@ -19,20 +20,38 @@ const services = Object.keys(allServices);
 
 console.log('services', services);
 
+const loadCerts = ({ caPath, certChainPath, keyPath }) =>
+  Promise.all([
+    fsPromises.readFile(caPath, 'UTF-8'),
+    fsPromises.readFile(certChainPath, 'UTF-8'),
+    fsPromises.readFile(keyPath, 'UTF-8'),
+  ]);
+
 const fetchWithCert = async (url, options) => {
-  const cert = fs.readFileSync(
-    process.env.CERT_CHAIN_PATH || '/etc/pki/tls/certs/client.crt',
-  );
-  const ca = fs.readFileSync(
-    process.env.CA_PATH || '/etc/pki/tls/certs/ca-bundle.crt',
-  );
-  const key = fs.readFileSync(
-    process.env.KEY_PATH || '/etc/pki/tls/private/client.key',
-  );
-  const agent = new Agent({ cert, ca, key });
-  return fetch(url, { agent, ...options });
-};
+  const caPath = process.env.CA_PATH || '/etc/pki/tls/certs/ca-bundle.crt';
+  const certChainPath =
+    process.env.CERT_CHAIN_PATH || '/etc/pki/tls/certs/client.crt';
+  const keyPath = process.env.KEY_PATH || '/etc/pki/tls/private/client.key';
+
+  const [ca, certChain, key] = await loadCerts({
+    caPath,
+    certChainPath,
+    keyPath,
+  });
 
+  return fetch(url, {
+    dispatcher: new Agent({
+      connect: {
+        secureContext: createSecureContext({
+          cert: certChain,
+          key,
+          ca,
+        }),
+      },
+    }),
+    ...options,
+  });
+};
 const timeTable = [];
 
 let csvContents;
```

Comment:
Just an aside but do we want to fix the lint warnings for `hasOwnProperty` whilst in here?

i.e using `Object.prototype.hasOwnProperty.call` for the `jsondata` to avoid conflicts with the prototype chain.

There's also a couple of `console.log` entries in this file that can possibly be removed

Reply:
I've removed those `hasOwnProperty` calls and replaced them with the `in` operator.

I think the `console.log`s are ok though as this is just a CLI script, so they are there for some output.